### PR TITLE
Update conditions details for tabular integrity checks

### PR DESCRIPTION
**Pull request by matanper**
Partial PR of #1338 

image

### CODE REVIEW

File: deepchecks/tabular/checks/data_integrity/is_single_value.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
         Returns
         -------
         CheckResult
-            value is a boolean if there was at least one column with only one unique,
+            value of result is a dict of all columns with number of uniques values as: {column: number_of_uniques, ...}
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/is_single_value.py:51]:
```suggestion
            value of result is a dict of all columns with number of unique values in format {column: number_of_uniques}
```

File: deepchecks/tabular/checks/data_integrity/is_single_value.py [hunk 2]
```diff
@@ -59,29 +59,32 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
 
         df = select_from_dataframe(df, self.columns, self.ignore_columns)
 
-        is_single_unique_value = (df.nunique(dropna=False) == 1)
+        num_unique_per_col = df.nunique(dropna=False)
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/is_single_value.py:62]: Make sure it doesn't count each none separately

File: deepchecks/tabular/checks/data_integrity/is_single_value.py [hunk 3]
```diff
@@ -59,29 +59,32 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
...
         name = 'Does not contain only a single value'
 
         def condition(result):
-            if result:
-                return ConditionResult(ConditionCategory.FAIL, f'Found columns with a single value: {result}')
-            return ConditionResult(ConditionCategory.PASS)
+            single_value_cols = [k for k, v in result.items() if v == 1]
+            if single_value_cols:
+                details = f'Found {len(single_value_cols)} columns with a single value out of {len(result)} ' \
+                          f'columns: {single_value_cols}'
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/is_single_value.py:85]: No need for total number of columns here in my opinion. Can be just "found X columns with single value: [...]"

File: deepchecks/tabular/checks/data_integrity/mixed_data_types.py [hunk 4]
```diff
@@ -58,7 +58,8 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
         Returns
         -------
         CheckResult
-            DataFrame with rows ('strings', 'numbers') for any column with mixed types.
+            value is a dict with column name as key and dict of 'strings' and 'numbers' ratio for any column with
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/mixed_data_types.py:61]:
```suggestion
            value is a dict where the key is the column name as key and the value is the ratio 'strings' and 'numbers' for any column with
```

File: deepchecks/tabular/checks/data_integrity/mixed_nulls.py [hunk 5]
```diff
@@ -69,7 +68,9 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
         Returns
         -------
         CheckResult
-            DataFrame with columns ('Column Name', 'Value', 'Count', 'Percentage') for any column which
+            Value is dict with columns as key, and dict of null values as value:
+            {column: {null_value: {count: x, percent: y}, ...}, ...}
+            display is DataFrame with columns ('Column Name', 'Value', 'Count', 'Percentage') for any column which
             have more than 1 null values.
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/mixed_nulls.py:74]:
```suggestion
            has more than 1 null values.
```

File: deepchecks/tabular/checks/data_integrity/mixed_nulls.py [hunk 6]
```diff
@@ -69,7 +68,9 @@ def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResul
         Returns
         -------
         CheckResult
-            DataFrame with columns ('Column Name', 'Value', 'Count', 'Percentage') for any column which
+            Value is dict with columns as key, and dict of null values as value:
+            {column: {null_value: {count: x, percent: y}, ...}, ...}
+            display is DataFrame with columns ('Column Name', 'Value', 'Count', 'Percentage') for any column which
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/mixed_nulls.py:73]:
```suggestion
            display is DataFrame with columns ('Column Name', 'Value', 'Count', 'Percentage') for any column that
```

File: deepchecks/tabular/checks/data_integrity/mixed_nulls.py [hunk 7]
```diff
@@ -164,18 +163,14 @@ def add_condition_different_nulls_not_more_than(self, max_allowed_null_types: in
...
-                num_nulls = len(nulls)
-                if num_nulls > max_allowed_null_types:
-                    not_passing_columns[column] = num_nulls
+            not_passing_columns = [k for k, v in result.items() if len(v) > max_allowed_null_types]
             if not_passing_columns:
-                return ConditionResult(ConditionCategory.FAIL,
-                                       'Found columns with amount of null types above threshold: '
-                                       f'{not_passing_columns}')
+                details = f'Found {len(not_passing_columns)} columns with amount of null types above threshold out ' \
+                          f'of {len(result)} columns: {not_passing_columns}'
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/mixed_nulls.py:169]: Again, total number of columns is not interesting in my opinion

File: deepchecks/tabular/checks/data_integrity/single_feature_contribution.py [hunk 8]
```diff
@@ -125,10 +125,12 @@ def condition(value: t.Dict[Hashable, float]) -> ConditionResult:
             }
 
             if failed_features:
-                message = f'Features with PPS above threshold: {failed_features}'
+                message = f'Found {len(failed_features)} features with PPS above threshold out of {len(value)} ' \
+                          f' features: {failed_features}'
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/single_feature_contribution.py:129]: Same comment about columns. Will stop commenting here

File: deepchecks/tabular/checks/data_integrity/string_length_out_of_bounds.py [hunk 9]
```diff
@@ -249,19 +247,17 @@ def compare_outlier_ratio(result: Dict):
...
             if not_passing_columns:
-                return ConditionResult(ConditionCategory.WARN,
-                                       f'Found columns with outliers ratio above threshold: '
-                                       f'{not_passing_columns}')
+                details = f'Found {len(not_passing_columns)} columns with outliers ratio above threshold out of ' \
+                          f'{len(result)} columns: {not_passing_columns}'
+                return ConditionResult(ConditionCategory.WARN, details)
             else:
-                return ConditionResult(ConditionCategory.PASS)
+                details = f'Passed for {len(result)} columns'
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/data_integrity/string_length_out_of_bounds.py:259]: Why not give any data?

File: tests/tabular/checks/integrity/string_mismatch_test.py [hunk 10]
```diff
@@ -154,6 +159,7 @@ def test_condition_percent_variants_no_more_than_pass():
     # Assert
     assert_that(result, has_items(
         equal_condition_result(is_pass=True,
+                               details='Passed for 1 string columns',
```
Comment (nirhutnik) [inline tests/tabular/checks/integrity/string_mismatch_test.py:162]: Here and in other places, we say "columns" even when the number is one.
Maybe add some sort of if so it won't be plural? I know it's annoying, and users will get why it's like that, but it's more professional like this.
Can also create a common method for that  - 

Suggestion:
def add_plural(num: int, postfix: str='s'):
    return postfix if num == 1 else ''

and then use as:
f'Found {num_columns} column{add_plural(num_columns)} ... '
and you can change the postfix into "es" where relevant.
Not great, but can change the experience a bit

**Review by nirhutnik — APPROVED** [review]

File: deepchecks/tabular/checks/data_integrity/string_length_out_of_bounds.py (hunk 9 above)
Comment (matanper) [reply deepchecks/tabular/checks/data_integrity/string_length_out_of_bounds.py:259]: if there are no outliers, there isn't any data to give

File: deepchecks/tabular/checks/data_integrity/is_single_value.py (hunk 3 above)
Comment (ItayGabbay) [reply deepchecks/tabular/checks/data_integrity/is_single_value.py:85]: I agree with Matan

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**